torch/csrc/distributed/c10d/ProcessGroupNCCL.cpp

```diff
@@ -997,11 +999,11 @@ void check_gpu_single_tensor(const at::Tensor& tensor) {
 
 // Check that all `tensors' have the same type and shape and are distributed
 // across distinct GPUs.
-void check_gpu_tensors(const std::vector<at::Tensor>& tensors) {
+int64_t check_gpu_tensors(const std::vector<at::Tensor>& tensors, bool same_device=false) {
```

Comment:
I feel like we are a bit overloading the responsibility of `check_gpu_tensors()` with this new `same_device` check. How about leaving `check_gpu_tensors()` as it is (maybe rename it to be more descriptive) and have a separate function (e.g. `check_tensors_on_same_device()`) that includes this new logic?

Reply:
sounds good to me.